### PR TITLE
[5.2][Events] Class for MailBeforeRendering event

### PR DESCRIPTION
**Pull request by Fedik**
### Summary of Changes

Add an event class for forgoten event `onMailBeforeRendering`

### Testing Instructions

In global configuration push "Send Test Mail" button.

### Actual result BEFORE applying this Pull Request

Email sent without error

### Expected result AFTER applying this Pull Request

Email sent without error

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed
- [ ] Pull Request link for manual.joomla.org: TBD
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by crommie** [comment]
I have tested this item :white_check_mark: successfully on c1ca9b20e2cc197e8e7cc97522ab4ae4f9afed5d<br /><br />No errors, all seems to work.

But: I'm working on a PBF server website. I didn't receive the Test e-mail so I'm not sure it all still works (yes, I did change the global email address). Since I didn't receive it before as well as after I think it might have something to do with something server-related. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43529">issues.joomla.org/tracker/joomla-cms/43529</a>.</sub>

**Comment by TLWebdesign** [comment]
I have tested this item :white_check_mark: successfully on c1ca9b20e2cc197e8e7cc97522ab4ae4f9afed5d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43529">issues.joomla.org/tracker/joomla-cms/43529</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43529">issues.joomla.org/tracker/joomla-cms/43529</a>.</sub>

File: libraries/src/Event/Mail/MailTemplateEvent.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    /**
+     * Pre-Setter for the subject argument.
+     *
+     * @param   MailTemplate  $value  The value to set
+     *
+     * @return  MailTemplate
+     *
+     * @since  __DEPLOY_VERSION__
+     */
+    protected function onSetSubject(MailTemplate $value): MailTemplate
```
Comment (Hackwar) [inline libraries/src/Event/Mail/MailTemplateEvent.php:76]: Could you explain what these methods are supposed to do? I don't see the use in this method and the following ones.

Comment (Fedik) [reply libraries/src/Event/Mail/MailTemplateEvent.php:76]: It is validation for the arguments. 
Try run following  code, and you will see :wink: 
```php
new BeforeRenderingMailTemplateEvent(
   'onMailBeforeRendering',
  ['templateId' => 'foobar', 'subject' => ['potato']]
);
```

**Comment by Hackwar** [comment]
Thanks for your contribution!

**Pull request merged**